# Project Overview

## Objective

This project aims to analyze the global data job market using publicly available job and salary data. The goal is to explore salary trendss, skill demand, remote work patterns, and geographic salary distribution within data-related roles.

By performing structured exploratory data analysis (EDA), this project will uncover meaningful insights that reflect current industry trends

## Why Job Market Analysis Matters

Understanding job market trends is essential for
* Identifying high-demand technical skills
* Estimating salary expectations across experience levels
* Comparing remote vs onsite opportunities
* Understanding geographic pay differences
* Making informed career development decisions

## Project Scope

This project focuses on
* Cleaning and preparing raw job market data
* Analyzing salary distributions
* Comparing salary trends by experience level
* Examining the impact of remote work on compensation
* Identifying the most in-demand skills
* Comparing median salaries across countries

## Business Questions

This analysis aims to answer the following questions.

1. What does the overall salary distribution look like for data-related roles?
2. How does salary vary by experience level (Entry, Mid, Senior, Executive)?
3. Do remote roles offer higher median salaries compared to onsite or hybrid roles?
4. Which technical skills are most frequently required in data job postings?
5. Which countries offer the highest median salaries for data roles?
6. Is there a noticeable salary difference based on company size?

### Environment Setup

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_style("whitegrid")

In [3]:
file_path = "../data/data_jobs.csv"
df = pd.read_csv(file_path)

df.head()

,work_year,experience_level,employment_type,job_title,salary,salary_currency,salary_in_usd,employee_residence,remote_ratio,company_location,company_size
0,2025,MI,FT,Research Scientist,208000,USD,208000,US,0,US,M
1,2025,MI,FT,Research Scientist,147000,USD,147000,US,0,US,M
2,2025,SE,FT,Research Scientist,173000,USD,173000,US,0,US,M
3,2025,SE,FT,Research Scientist,117000,USD,117000,US,0,US,M
4,2025,MI,FT,AI Engineer,100000,USD,100000,US,100,US,M


## Data Understanding

In [4]:
df.shape

(93597, 11)

This dataset contains 93597 rows and 11 columns.

In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 93597 entries, 0 to 93596
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   work_year           93597 non-null  int64
 1   experience_level    93597 non-null  str  
 2   employment_type     93597 non-null  str  
 3   job_title           93597 non-null  str  
 4   salary              93597 non-null  int64
 5   salary_currency     93597 non-null  str  
 6   salary_in_usd       93597 non-null  int64
 7   employee_residence  93597 non-null  str  
 8   remote_ratio        93597 non-null  int64
 9   company_location    93597 non-null  str  
 10  company_size        93597 non-null  str  
dtypes: int64(4), str(7)
memory usage: 7.9 MB


Most categorical variables are stored as object types, while salary-related columns are numeric.

In [6]:
df.describe()

,work_year,salary,salary_in_usd,remote_ratio
count,93597.000000,9.359700e+04,93597.000000,93597.000000
mean,2024.086434,1.623541e+05,157547.696774,21.455816
std,0.641449,2.221425e+05,73649.113729,40.954704
min,2020.000000,1.400000e+04,15000.000000,0.000000
25%,2024.000000,1.062600e+05,106250.000000,0.000000
50%,2024.000000,1.470000e+05,146232.000000,0.000000
75%,2024.000000,1.990000e+05,198000.000000,0.000000
max,2025.000000,3.040000e+07,800000.000000,100.000000


The salary_in_usd column shows a wide range, with a noticeable difference between median and maximum values

### Column Description

* work_year - Year the salary was recorded
* experience_level - Employee seniority level
* employment_type - Full-time, part-time, contract, Freelance
* job_title - Specific job role
* salary - Salary in original currency
* salary_currency - Currency type
* salary_in_usd - Salary standardized to USD
* employee_residence - Country of employee
* remote_ratio - Percentage of remote work
* company_location - company headquarters location
* company_size - Small(S), medium (M), Large (L)

In [8]:
df['experience_level'].unique()

<StringArray>
['MI', 'SE', 'EN', 'EX']
Length: 4, dtype: str

### Experience Level Mapping

* EN - Entry level
* MI - Mid level
* SE - Senior level
* EX - Executive level

In [9]:
df['employment_type'].unique()

<StringArray>
['FT', 'PT', 'CT', 'FL']
Length: 4, dtype: str

* FT - Full time
* PT - Part time
* CT - Contract
* FL - Freelance

In [10]:
df['remote_ratio'].unique()

array([  0, 100,  50])

In [11]:
df['company_size'].unique()

<StringArray>
['M', 'L', 'S']
Length: 3, dtype: str

* M - Medium
* L - Large
* S - Small

In [12]:
df.isnull().sum()

work_year             0
experience_level      0
employment_type       0
job_title             0
salary                0
salary_currency       0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

In [13]:
df.duplicated().sum()

np.int64(46960)

The dataset contains 46960 duplicate records. In the cleaning stage these will be removed.

For consistency and comparability across countries, salary_in_usd will be used as the primary salary metric in this analysis.

In [ ]:
df['remote_ratio'].value_counts